core/src/main/java/com/linecorp/armeria/internal/annotation/ObjectCollectingUtil.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.annotation;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+/**
+ * A utility class which provides functions for collecting objects published from a {@link Publisher} or
+ * {@link Stream}.
+ */
+final class ObjectCollectingUtil {
+
+    /**
+     * The {@link Class} instance of {@code reactor.core.publisher.Mono} of
+     * <a href="https://projectreactor.io/">Project Reactor</a>.
+     */
+    private static final Class<?> MONO_CLASS;
+
+    static {
+        Class<?> mono = null;
+        try {
+            mono = Class.forName("reactor.core.publisher.Mono");
```

Comment:
Please specify the class loader, e.g. `ObjectCollectingUtil.class.getClassLoader()`.